compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/schema/generator/SDLFileGenerator.java

```diff
@@ -0,0 +1,391 @@
+            return getFormattedString(NON_NULL_FORMAT, createFieldType(type.getOfType()));
+        } else if (type.getKind().equals(TypeKind.LIST)) {
+            return getFormattedString(LIST_FORMAT, createFieldType(type.getOfType()));
+        } else {
+            return type.getName();
+        }
+    }
+
+    private String createArgType(InputValue arg) {
+        if (arg.getDefaultValue() == null) {
+            return createFieldType(arg.getType());
+        } else {
+            return getFormattedString(ARGS_TYPE_FORMAT, createFieldType(arg.getType()), arg.getDefaultValue());
+        }
+    }
+
+    private String createInterfaceImplements(Type type) {
+        List<String> interfaces = new ArrayList<>();
+        if (type.getInterfaces().isEmpty()) {
+            return EMPTY_STRING;
+        }
+        for (Type interfaceType : type.getInterfaces()) {
+            interfaces.add(interfaceType.getName());
+        }
+        return getFormattedString(IMPLEMENT_FORMAT, String.join(AND_SIGN, interfaces));
+    }
+
+    private String createFieldDescription(String description) {
+        if (description == null || description.isEmpty()) {
+            return EMPTY_STRING;
+        } else {
+            List<String> subStrings = new ArrayList<>();
+            for (String subString : description.split(LINE_SEPARATOR)) {
+                subStrings.add(getFormattedString(COMMENT_FORMAT, INDENTATION, subString));
+            }
+            return getFormattedString(DESC_FORMAT, String.join(LINE_SEPARATOR, subStrings));
+        }
+    }
+
+    private String createDirectiveLocation(List<DirectiveLocation> location) {
+        List<String> locations = new ArrayList<>();
+        for (DirectiveLocation directiveLocation : location) {
+            locations.add(directiveLocation.name());
+        }
+        return String.join(PIPE_SIGN, locations);
+    }
+
+    private String createDeprecate(EnumValue enumValue) {
+        if (enumValue.isDeprecated()) {
+            if (enumValue.getDeprecationReason() != null) {
+                return getFormattedString(DEPRECATE_FORMAT, DEPRECATE, enumValue.getDeprecationReason());
+            }
+            return DEPRECATE;
+        }
+        return EMPTY_STRING;
+    }
+
+    private String createDeprecate(Field field) {
+        if (field.isDeprecated()) {
+            if (field.getDeprecationReason() != null) {
+                return getFormattedString(DEPRECATE_FORMAT, DEPRECATE, field.getDeprecationReason());
+            }
+            return DEPRECATE;
+        }
+        return EMPTY_STRING;
+    }
+
+    private Boolean isIntrospectionType(Type type) {
+        for (IntrospectionType value : IntrospectionType.values()) {
+            if (value.getName().equals(type.getName())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Boolean isBuiltInScalarType(Type type) {
+        for (ScalarType value : ScalarType.values()) {
```

Comment:
The `ScalarType` contains the `Decimal` type and also the `Upload` type. These aren't built-in scalar types. They should be added to the schema if they are present in the service. We should find a better approach for this.

Reply:
When I meant a `Better Approach`, I thought of a more elegant solution that special-casing these two types. Instead of comparing strings, we may at least improve this to check the type kind. Or maybe keep a separate flag to check whether it is an inbuilt type.

Reply:
Do you mean a change on the schema object generation side?

Reply:
Added a separate flag into `ScalarType` enum.